### PR TITLE
AB#2901: mock country list and province list

### PR DESCRIPTION
**Pull request by leiye29**
### Description

1. mock country list
2. mock province list
3. add function into lookup-service to retrieve country list and province list

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/mocks/db.ts [hunk 1]
```diff
@@ -44,6 +44,17 @@ const db = factory({
...
     id: primaryKey(faker.string.uuid),
   },
+  country: {
+    code: primaryKey(String),
+    nameEn: String,
+    nameFr: String,
+  },
+  region: {
+    code: primaryKey(String),
+    countryCode: String,
```
Comment (nicholas-ly) [inline frontend/app/mocks/db.ts:54]: Consider using a [One-to-One relationship](https://github.com/mswjs/data?tab=readme-ov-file#one-to-one).

File: frontend/app/services/lookup-service.server.ts [hunk 2]
```diff
@@ -60,9 +73,56 @@ function createLookupService() {
...
+    });
+
+    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+  }
+
   return {
     getAllPreferredLanguages,
     getPreferredLanguage,
+    getCountries,
+    getRegions,
```
Comment (nicholas-ly) [inline frontend/app/services/lookup-service.server.ts:125]: **Nitpick:** To be consistent with what currently exists for preferred languages, consider renaming these functions `getAllCountries` and `getAllRegions`.

File: frontend/app/services/lookup-service.server.ts [hunk 3]
```diff
@@ -9,6 +9,19 @@ const preferredLanguageSchema = z.object({
...
+  code: z.string(),
+  nameEn: z.string().optional(),
+  nameFr: z.string().optional(),
+});
+
+const regionSchema = z.object({
+  code: z.string(),
+  countryCode: z.string(),
+  nameEn: z.string().optional(),
+  nameFr: z.string().optional(),
```
Comment (nicholas-ly) [inline frontend/app/services/lookup-service.server.ts:22]: I don't see why `nameEn` and `nameFr` should be optional. The APIs we are dealing with should be able to provide countries and regions, in both official languages.

File: frontend/app/services/lookup-service.server.ts [hunk 4]
```diff
@@ -60,9 +73,56 @@ function createLookupService() {
...
+   *
+   * @returns returns province list
+   */
+  async function getRegions() {
+    const url = `${INTEROP_API_BASE_URI}/lookups/regions`;
+    const response = await fetch(url);
+
+    const regionListSchema = z.array(regionSchema);
+    if (response.ok) return regionListSchema.parse(await response.json());
+    if (response.status === 404) return null;
```
Comment (nicholas-ly) [inline frontend/app/services/lookup-service.server.ts:108]: This line (`if (response.status === 404) return null;`) can be removed.

The API should return an empty collection if no regions are found. If there the response results in a 404, we will throw instead.

File: frontend/app/mocks/db.ts (hunk 1 above)
Comment (leiye29) [reply frontend/app/mocks/db.ts]: fixed

File: frontend/app/services/lookup-service.server.ts (hunk 2 above)
Comment (leiye29) [reply frontend/app/services/lookup-service.server.ts]: fixed

File: frontend/app/services/lookup-service.server.ts (hunk 3 above)
Comment (leiye29) [reply frontend/app/services/lookup-service.server.ts]: fixed

File: frontend/app/services/lookup-service.server.ts (hunk 4 above)
Comment (leiye29) [reply frontend/app/services/lookup-service.server.ts]: removed

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**